### ISSUE TITLE
Error trying to write to HOME (dhcp component)

### ISSUE THREAD

**Issue by tribut** [tribut/homeassistant-docker-venv#20]
The dhcp component tries to write to $HOME, which is `/root` and therefore fails:

~~~~
2021-04-09 11:19:39 ERROR (MainThread) [homeassistant.setup] Setup failed for dhcp: unknown error                                                            
Traceback (most recent call last):                                                                                                                            
  File "/usr/src/homeassistant/homeassistant/setup.py", line 190, in _async_setup_component                                                                  
    component = integration.get_component()                                                                                                                  
  File "/usr/src/homeassistant/homeassistant/loader.py", line 477, in get_component                                                                           
    cache[self.domain] = importlib.import_module(self.pkg_path)                                                                      
  File "/usr/local/lib/python3.8/importlib/__init__.py", line 127, in import_module                                                                          
    return _bootstrap._gcd_import(name[level:], package, level)                                                                                               
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import                                                                            
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load                                                                                          
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked                                                                                  
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked                                                                                           
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module                                                                                     
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed                                                                                
  File "/usr/src/homeassistant/homeassistant/components/dhcp/__init__.py", line 11, in <module>                                                               
    from aiodiscover import DiscoverHosts
  File "/usr/local/lib/python3.8/site-packages/aiodiscover/__init__.py", line 11, in <module>
    from .discovery import DiscoverHosts  # noqa: F401
  File "/usr/local/lib/python3.8/site-packages/aiodiscover/discovery.py", line 5, in <module>
    from async_dns.core import REQUEST, DNSMessage, RandId, Record, types
  File "/usr/local/lib/python3.8/site-packages/async_dns/__init__.py", line 1, in <module>
    from .core import *
  File "/usr/local/lib/python3.8/site-packages/async_dns/core/__init__.py", line 5, in <module>
    from .config import *
  File "/usr/local/lib/python3.8/site-packages/async_dns/core/config/__init__.py", line 2, in <module>
    from .root import *
  File "/usr/local/lib/python3.8/site-packages/async_dns/core/config/root.py", line 16, in <module>
    os.makedirs(CONFIG_DIR, exist_ok=True)
  File "/usr/local/lib/python3.8/os.py", line 213, in makedirs
    makedirs(head, exist_ok=exist_ok)
  File "/usr/local/lib/python3.8/os.py", line 223, in makedirs
    mkdir(name, mode)
PermissionError: [Errno 13] Permission denied: '/root/.config'
~~~~

**Comment by fanaticDavid** [comment]
I am glad to have found this issue! After updating HA to version 2021.4.x, the Broadlink integration stopped working for me with an error like the one above in the log file. I couldn't figure out why, because I couldn't find any reports in the HA Core repository. I tried removing the integration to add it back, but it fails to load the config flow.

Then a lightbulb turned on in my brain and I remembered that I use this script to run the HA docker container as a non-root user, which could explain the permission issue. Looking forward to the fix! 👍
